lib/igniter/test.ex

```diff
@@ -41,45 +40,50 @@ defmodule Igniter.Test do
   end
 
   @doc """
-  IO.puts the current igniter diff, and returns the igniter
+  Print the current `igniter` diff, returning the `igniter`.
+
+  This is primarily used for debugging purposes.
 
   ## Options
 
-  * `label` - A label to put before the diff.
-  * `only` - File path(s) to only show the diff for
+  * `:label` - A label to print before the diff
+  * `:only` - Only print the diff for this file or files
   """
   @spec puts_diff(Igniter.t(), opts :: Keyword.t()) :: Igniter.t()
   def puts_diff(igniter, opts \\ []) do
-    igniter.rewrite.sources
-    |> Map.values()
-    |> then(fn sources ->
-      if opts[:only] do
-        Enum.filter(sources, &(&1.path in List.wrap(opts[:only])))
-      else
-        sources
-      end
-    end)
-    |> Igniter.diff()
-    |> String.trim()
-    |> case do
-      "" ->
-        if opts[:label] do
-          IO.puts("#{opts[:label]} No changes!")
-        else
-          IO.puts("No changes!")
-        end
+    only = Keyword.get(opts, :only)
+    label = Keyword.get(opts, :label, "")
+    prefix = if label == "", do: "", else: "#{label}: "
 
-      diff ->
-        if opts[:label] do
-          IO.puts("#{opts[:label]}\n\n#{diff}")
-        else
-          IO.puts(diff)
-        end
+    case igniter |> diff(only: only, color?: true) |> String.trim() do
+      "" -> IO.puts("#{prefix}no changes!")
+      diff -> IO.puts("#{prefix}\n\n#{diff}")
```

Comment:
Oops, this will actually add 2 extra newlines even if you don't specify a label. Will fix.

Reply:
Fixed.